Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm

```diff
@@ -47,20 +50,56 @@
 }
 
 #if HAVE(WEB_TRANSPORT)
-static RetainPtr<nw_parameters_t> createParameters(WebCore::SecurityOriginData&& origin)
+
+static void didReceiveServerTrustChallenge(Ref<NetworkConnectionToWebProcess>&& connectionToWebProcess, URL&& url, WebKit::WebPageProxyIdentifier&& pageID, WebCore::SecurityOriginData&& topOrigin, sec_trust_t trust, sec_protocol_verify_complete_t completion)
+{
+    const char * host = url.host().utf8().data();
+    uint16_t port = url.port() ? *url.port() : 443;
+    RetainPtr secTrust = adoptCF(sec_trust_copy_ref(trust));
+    RetainPtr protectionSpace = adoptNS([[NSURLProtectionSpace alloc] initWithHost:[NSString stringWithUTF8String:host] port:port protocol:NSURLProtectionSpaceHTTPS realm:nil authenticationMethod:NSURLAuthenticationMethodServerTrust]);
+    [protectionSpace _setServerTrust:secTrust.get()];
+
+    id<NSURLAuthenticationChallengeSender> sender = nil;
+    RetainPtr challenge = adoptNS([[NSURLAuthenticationChallenge alloc] initWithProtectionSpace:protectionSpace.get() proposedCredential:nil previousFailureCount:0 failureResponse:nil error:nil sender:sender]);
+
+    auto challengeCompletionHandler = [completion = makeBlockPtr(completion), secTrust = WTFMove(secTrust)] (AuthenticationChallengeDisposition disposition, const WebCore::Credential& credential) {
+        switch (disposition) {
+        case AuthenticationChallengeDisposition::UseCredential: {
+            if (!credential.isEmpty())
+                completion(true);
+        }
+        FALLTHROUGH;
+        case AuthenticationChallengeDisposition::RejectProtectionSpaceAndContinue:
+        case AuthenticationChallengeDisposition::PerformDefaultHandling: {
+            OSStatus status = SecTrustEvaluateAsyncWithError(secTrust.get(), dispatch_get_main_queue(), makeBlockPtr([completion = WTFMove(completion)](SecTrustRef trustRef, bool result, CFErrorRef error) {
+                completion(result);
+            }).get());
+            if (status != errSecSuccess)
+                completion(false);
```

Comment:
This is a use-after-move.  This will always dereference null because completion has been moved from.  In order for something like this to work we need to keep a strong reference to the block and have a strong reference captured in the lambda/block given to SecTrustEvaluateAsyncWithError.